src/qt/rpcconsole.cpp

```diff
@@ -1192,31 +1193,15 @@ void RPCConsole::updateDetailWidget()
         ui->peerPermissions->setText(permissions.join(" & "));
     }
     ui->peerMappedAS->setText(stats->nodeStats.m_mapped_as != 0 ? QString::number(stats->nodeStats.m_mapped_as) : ts.na);
-
-    // This check fails for example if the lock was busy and
-    // nodeStateStats couldn't be fetched.
-    if (stats->fNodeStateStatsAvailable) {
-        ui->peerServices->setText(GUIUtil::formatServicesStr(stats->nodeStateStats.their_services));
-        // Sync height is init to -1
-        if (stats->nodeStateStats.nSyncHeight > -1) {
-            ui->peerSyncHeight->setText(QString("%1").arg(stats->nodeStateStats.nSyncHeight));
-        } else {
-            ui->peerSyncHeight->setText(ts.unknown);
-        }
-        // Common height is init to -1
-        if (stats->nodeStateStats.nCommonHeight > -1) {
-            ui->peerCommonHeight->setText(QString("%1").arg(stats->nodeStateStats.nCommonHeight));
-        } else {
-            ui->peerCommonHeight->setText(ts.unknown);
-        }
-        ui->peerHeight->setText(QString::number(stats->nodeStateStats.m_starting_height));
-        ui->peerPingWait->setText(GUIUtil::formatPingTime(stats->nodeStateStats.m_ping_wait));
-        ui->peerAddrRelayEnabled->setText(stats->nodeStateStats.m_addr_relay_enabled ? ts.yes : ts.no);
-        ui->peerAddrProcessed->setText(QString::number(stats->nodeStateStats.m_addr_processed));
-        ui->peerAddrRateLimited->setText(QString::number(stats->nodeStateStats.m_addr_rate_limited));
-        ui->peerRelayTxes->setText(stats->nodeStateStats.m_relay_txs ? ts.yes : ts.no);
-    }
-
+    ui->peerServices->setText(GUIUtil::formatServicesStr(stats->nodeStateStats.m_services));
```

Comment:
What about making `stats->nodeStateStats.m_services` `std::optional` and here do `x.has_value() ? ... : "Unknown")`? Otherwise we seem to be abusing a real value `NODE_NONE` (no services) as a dummy "we don't know/has no value".

Reply:
Done